modules/core/src/main/java/org/apache/ignite/internal/util/nio/ssl/GridNioSslHandler.java

```diff
@@ -56,6 +56,9 @@ class GridNioSslHandler extends ReentrantLock {
     /** Handshake duration threshold; warning is logged when exceeded. */
     private static final long LONG_HANDSHAKE_THRESHOLD_MS = 1000;
 
+    /** Handshake timeout; exception is thrown when exceeded. */
+    private static final long HANDSHAKE_TIMEOUT_MS = 3000;
```

Comment:
This looks like a hard limit, what about making it configurable via java properties just in case?

Reply:
Good point. Added `IGNITE_SSL_HANDSHAKE_TIMEOUT` system property.

Reply:
Good point, dome.